pkg/urpc/urpc.go

```diff
@@ -89,6 +89,35 @@ func (f *FilePayload) setFilePayload(fs []*os.File) {
 	f.Files = fs
 }
 
+// ReadonlyFilePayload is the same as FilePayload, except that it does not
+// export the underlying files array. Using an unexported type alias, it can be
+// embedded without being assignable directly outside a package. This is useful
+// for maintaining invariants over the files and other struct fields.
+type ReadonlyFilePayload struct {
+	files []*os.File
+}
+
+// NewPrivateFilePayload creates the readonly payload.
+func NewPrivateFilePayload(fs []*os.File) *ReadonlyFilePayload {
```

Comment:
Since the type is `ReadonlyFilePayload`, should the constructor be named `NewReadonlyFilePayload()` for consistency?  Or they could be `PrivateFilePayload` / `NewPrivateFilePayload()`.

Reply:
I have removed the `ReadonlyFilePayload` entirely now. This is explained in more detail below.